.github/workflows/check.yaml

```diff
@@ -0,0 +1,32 @@
+name: Check workflow running linter, unit and functional tests
+
+on:
+  workflow_call:
+  pull_request:
+    types: [opened, synchronize, reopened]
+    branches: [master, main]
+    paths-ignore:
+      - "**.md"
+      - "**.rst"
+
+jobs:
+  lint-unit:
+    uses: canonical/bootstack-actions/.github/workflows/lint-unit.yaml@v2
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.8", "3.10"]
+    with:
+      python-version: ${{ matrix.python-version }}
+      tox-version: "<4"
+
+  func:
+    uses: canonical/bootstack-actions/.github/workflows/func.yaml@v2
+    needs: lint-unit
+    with:
+      command: tox -e func
+      juju-channel: "3.1/stable"
```

Comment:
Since charm-nginx needs to support both juju 2.9 and 3.1, we would need to run functional tests for both juju-channels as well. Could you create a matrix strategy here to call both values?

Reply:
charm-nginx belongs to case2. We can run 1 channel only.

Reply:
This is different though. Charms in Launchpad will have weekly 2.9 functest runs to confirm their compatibility; but in GitHub, unless we test also test for 2.9 in workflow for new changes, we wouldn't know if the code base is still compatible (creating weekly workflow per repo requires more work).

Plus in Github, we don't need to worry about CI verification polluting the PR (in contrast to Laucnpad's CI vote), so I don't see why we don't want to test for both channels.

Reply:
Something to note here is that charm-nginx currently does not have any functional tests whatsoever. `tox -e func` just prints out that the functional tests are under development. So in this case, would we still need to make changes to the CI in order to also perform "functional tests" for juju 2.9?

Reply:
Please skip it. It's a charm nearly with low activity.